File: sdks/sdk-java/src/main/java/com/openpay/sdk/exceptions/ApiException.java
package com.openpay.sdk.exceptions;

public class ApiException extends OpenPayException {
    private final String code;
    private final int httpStatus;

    public ApiException(String code, String message, int httpStatus) {
        super(message + " (" + code + ", HTTP " + httpStatus + ")");
        this.code = code;
        this.httpStatus = httpStatus;
    }

    public String getCode() { return code; }
    public int getHttpStatus() { return httpStatus; }
    public boolean isNotFound() { return httpStatus == 404; }
    public boolean isAuthError() { return httpStatus == 401; }
}

File: sdks/sdk-php/src/Exceptions/ApiException.php
<?php

declare(strict_types=1);

namespace OpenPay\Exceptions;

class ApiException extends OpenPayException
{
    public function __construct(
        public readonly string $errorCode,
        string $message,
        public readonly int $httpStatus,
    ) {
        parent::__construct("{$message} ({$errorCode}, HTTP {$httpStatus})", $httpStatus);
    }
}
